### PR TITLE
Improved installation

### PR DESCRIPTION
**Pull request by noamnelke**
- Check go ver is 1.11+.
- Install deps ahead of relying on them.

_This is currently bash only._

### CODE REVIEW

**Comment by noamnelke** [comment]
Fixes https://github.com/spacemeshos/go-spacemesh/issues/440

**Comment by noamnelke** [comment]
@antonlerner this is currently bash only -- do you want to enlist the community in checking minimum go version on Windows?

**Comment by antonlerner** [comment]
If it's not a big deal, I'd rather we solve this internally, we had some dev's struggling with this one. 
Also, and regardless, we need to check that this script runs on linux (ubuntu) and mac as well - this is the most important goal of all these fixes - to allow our users to setup our env easily.

**Comment by noamnelke** [comment]
@antonlerner The installation works on Mac (tested locally) and on Ubuntu (tested on Travis).

It's supposed to fail fast with an old version of Go and I've tested this [on Travis](https://travis-ci.org/spacemeshos/go-spacemesh/jobs/483414484).

Making the Go version check work on Windows is not trivial. I've used `sed` to parse the version string, and bash scripting to compare to the expected minimum version. I have no experience with Windows batch file programming and no Windows machine to test on here. I can learn some Windows scripting and test tomorrow in the office, or install a Windows VM, but this is a non-critical, nice to have, part of the installation that should just help with debugging (help developers understand that they have a version of Go that's too old). Implementing it for Windows seems not worth the effort to me.

@zalmen - wdyt?

File: setup_env.sh [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #!/bin/bash -e
+# Ensure we use Go version 1.11+
+read major minor patch <<< $(go version | sed 's/go version go\([0-9]*\)\.\([0-9]*\)\.\(.*\) .*/\1 \2 \3/')
```
Comment (antonlerner) [inline setup_env.sh]: is sed found in a clean ubuntu setup?

Comment (noamnelke) [reply setup_env.sh]: Looks that way.

The [UNIX specification](https://en.wikipedia.org/wiki/List_of_Unix_commands) declares that it's mandatory:
image

A random [dude on AskUbuntu](https://askubuntu.com/a/957815) puts it nicely:

> `sed` is a utility that programs are generally allowed to assume exists and to rely on themselves. As you noticed, the scripts that run when installing, removing, or upgrading software can use `sed`.

**Comment by noamnelke** [comment]
@antonlerner I've added Windows support. I'm relying on PowerShell, which is installed by default since Windows 7.

**Review by antonlerner — APPROVED** [review]

**Pull request merged**